airbyte-integrations/connectors/destination-pubsub/src/main/java/io/airbyte/integrations/destination/pubsub/PubsubDestination.java

```diff
@@ -0,0 +1,91 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.destination.pubsub;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.pubsub.v1.TopicAdminClient;
+import com.google.cloud.pubsub.v1.TopicAdminSettings;
+import com.google.common.base.Charsets;
+import com.google.pubsub.v1.TopicName;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.integrations.BaseConnector;
+import io.airbyte.integrations.base.AirbyteMessageConsumer;
+import io.airbyte.integrations.base.Destination;
+import io.airbyte.integrations.base.IntegrationRunner;
+import io.airbyte.protocol.models.AirbyteConnectionStatus;
+import io.airbyte.protocol.models.AirbyteConnectionStatus.Status;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import java.io.ByteArrayInputStream;
+import java.util.function.Consumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PubsubDestination extends BaseConnector implements Destination {
+
+  static final String CONFIG_TOPIC_ID = "topic_id";
+  static final String CONFIG_PROJECT_ID = "project_id";
+  static final String CONFIG_CREDS = "credentials_json";
+  static final String STREAM = "_stream";
+  static final String NAMESPACE = "_namespace";
+  private static final Logger LOGGER = LoggerFactory.getLogger(PubsubDestination.class);
+
+  public static void main(String[] args) throws Exception {
+    new IntegrationRunner(new PubsubDestination()).run(args);
+  }
+
+  @Override
+  public AirbyteConnectionStatus check(JsonNode config) {
+    try {
+      final String projectId = config.get(CONFIG_PROJECT_ID).asText();
+      final String topicId = config.get(CONFIG_TOPIC_ID).asText();
+      final String credentialsString =
+          config.get(CONFIG_CREDS).isObject() ? Jsons.serialize(config.get(CONFIG_CREDS))
+              : config.get(CONFIG_CREDS).asText();
+      final ServiceAccountCredentials credentials = ServiceAccountCredentials
+          .fromStream(new ByteArrayInputStream(credentialsString.getBytes(Charsets.UTF_8)));
+
+      TopicAdminClient adminClient = TopicAdminClient
+          .create(TopicAdminSettings.newBuilder().setCredentialsProvider(
+              FixedCredentialsProvider.create(credentials)).build());
+      TopicName topicName = TopicName.of(projectId, topicId);
+      adminClient.getTopic(topicName);
```

Comment:
is there a way to also verify that the given svc user can publish messages? Not strictly required but would be a nice UX improvement

Reply:
agreed that sending an actual message is not good here :)

Reply:
Found a way to test/check iam permissions. Let me know if that will do